packages/autoloader/src/class-plugins-handler.php

```diff
@@ -2,128 +2,159 @@
  */
 class Plugins_Handler {
+	/**
+	 * The transient key for plugin paths.
+	 */
+	const TRANSIENT_KEY = 'jetpack_autoloader_plugin_paths';
 
 	/**
-	 * Returns an array containing the paths of all active plugins and all known activating plugins.
+	 * The locator for finding plugins in different locations.
 	 *
-	 * @return array An array of plugin paths as strings or an empty array.
+	 * @var Plugin_Locator
 	 */
-	public function get_all_active_plugins_paths() {
-		global $jetpack_autoloader_activating_plugins_paths;
-
-		$active_plugins_paths     = $this->get_active_plugins_paths();
-		$multisite_plugins_paths  = $this->get_multisite_plugins_paths();
-		$activating_plugins_paths = $this->get_plugins_activating_via_request();
-
-		return array_unique(
-			array_merge(
-				$active_plugins_paths,
-				$activating_plugins_paths,
-				$multisite_plugins_paths,
-				(array) $jetpack_autoloader_activating_plugins_paths
-			)
-		);
-	}
+	private $plugin_locator;
 
 	/**
-	 * Returns an array containing the paths of the active sitewide plugins in a multisite environment.
+	 * The processor for transforming cached paths.
 	 *
-	 * @return array The paths of the active sitewide plugins or an empty array.
+	 * @var Path_Processor
 	 */
-	protected function get_multisite_plugins_paths() {
-		$plugin_slugs = is_multisite()
-			? array_keys( get_site_option( 'active_sitewide_plugins', array() ) )
-			: array();
-
-		$plugin_slugs = array_filter( $plugin_slugs, array( $this, 'is_directory_plugin' ) );
-		return array_map( array( $this, 'create_plugin_path' ), $plugin_slugs );
-	}
+	private $path_processor;
 
 	/**
-	 * Returns an array containing the paths of the currently active plugins.
+	 * The constructor.
 	 *
-	 * @return array The active plugins' paths or an empty array.
+	 * @param Plugin_Locator $plugin_locator The locator for finding active plugins.
+	 * @param Path_Processor $path_processor The processor for transforming cached paths.
 	 */
-	protected function get_active_plugins_paths() {
-		$plugin_slugs = (array) get_option( 'active_plugins', array() );
-		$plugin_slugs = array_filter( $plugin_slugs, array( $this, 'is_directory_plugin' ) );
-		return array_map( array( $this, 'create_plugin_path' ), $plugin_slugs );
+	public function __construct( $plugin_locator, $path_processor ) {
+		$this->plugin_locator = $plugin_locator;
+		$this->path_processor = $path_processor;
 	}
 
 	/**
-	 * Adds the plugin directory from the WP_PLUGIN_DIR constant to the plugin slug.
+	 * Gets the directory of the current plugin.
 	 *
-	 * @param string $plugin_slug The plugin slug.
+	 * @return string
 	 */
-	private function create_plugin_path( $plugin_slug ) {
-		$plugin_dir = str_replace( '\\', '/', WP_PLUGIN_DIR );
-		return trailingslashit( $plugin_dir ) . substr( $plugin_slug, 0, strrpos( $plugin_slug, '/' ) );
+	public function get_current_plugin() {
+		// Escape from `vendor/__DIR__` to plugin directory.
+		return dirname( dirname( __DIR__ ) );
```

Comment:
I think we need to normalize the current plugin's path with `str_replace( '\\', '/', dirname( dirname( __DIR__ ) );` here.

Reply:
I've gone ahead and moved this functionality into the `Plugin_Locator` since it seems to fit some of the dependencies already there as well as the general theme of the class. It's now normalized correctly as a result!